templates/terraform/decoders/kms_key_ring.go.erb

```diff
@@ -0,0 +1,21 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2018 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+	config := meta.(*Config)
+	res["self_link"] = d.Get("name").(string)
```

Comment:
I think this should actually be `res["self_link"] = res["name"]` since at this time `d.Get("name")` only has the short form, and for the self_link we need the full form, which is returned from the Read. Can you confirm?

Reply:
Yup, that sounds right. What about the lines right underneath this- are those doing anything?

Reply:
The idea was to set the Id on import before we had import_format. Gone now.

I had to remove `ignore_read: true` from `name`.  If all fields are ignored on Read, as it was the case before, the linter will complain that the returned `res` from the Decoder is innefectual:
```
google/resource_kms_key_ring.go:124:2: ineffectual assignment to `res` (ineffassign)
	res, err = resourceKmsKeyRingDecoder(d, meta, res)
```
So I now set it in the decoder to the short form and let the Read add that to the state after the decoder returns.